### PR TITLE
Flow update

### PR DESCRIPTION
**Pull request by hebelsan**
**How to categorize this PR?**

/area quality
/kind cleanup
/platform aws

**What this PR does / why we need it**:
Updates the aws infra controller

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by hebelsan** [comment]
Needs change because integration tests are failing!

**Review by kon-angelo — COMMENTED** [review]
/lgtm

**Review by AndreasBurger — CHANGES_REQUESTED** [review]
There is a common bug in the flow-reconciliation (for an example, see https://github.com/gardener/gardener-extension-provider-openstack/pull/824).
We should not merge this unless we're sure it is fixed

**Comment by kon-angelo** [comment]
@AndreasBurger should be fixed with 05a28018db6c4dc22ef383c1e34b3ddb9326115b

File: pkg/controller/infrastructure/flow_reconciler.go
Comment (AndreasBurger) [inline pkg/controller/infrastructure/flow_reconciler.go:95]: Debug/dev leftover?

File: pkg/controller/infrastructure/flow_reconciler.go
Comment (AndreasBurger) [inline pkg/controller/infrastructure/flow_reconciler.go:51]: Debug/dev leftover?

File: pkg/controller/infrastructure/flow_reconciler.go
Comment (AndreasBurger) [inline pkg/controller/infrastructure/flow_reconciler.go:196]: The initial logging on migration is arguably useful. This one does look more like a leftover to me though.

File: pkg/controller/infrastructure/infraflow/context.go
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/context.go:165]: if this is not needed, just rm

File: pkg/controller/infrastructure/infraflow/context.go
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/context.go:276]: nit: we use `vpcID`, shouldn't it be `arnIAMRole` then?

feel free to disagree

File: pkg/controller/infrastructure/flow_reconciler.go
Comment (hebelsan) [reply pkg/controller/infrastructure/flow_reconciler.go]: reduced log verbosity

File: pkg/controller/infrastructure/flow_reconciler.go
Comment (hebelsan) [reply pkg/controller/infrastructure/flow_reconciler.go]: lowered log verbosity

File: pkg/controller/infrastructure/infraflow/context.go
Comment (hebelsan) [reply pkg/controller/infrastructure/infraflow/context.go]: agree, changed to `arnIAMRole`

File: pkg/controller/infrastructure/infraflow/context.go
Comment (hebelsan) [reply pkg/controller/infrastructure/infraflow/context.go]: removed because not needed

File: pkg/controller/infrastructure/infraflow/utils.go
Comment (hebelsan) [inline pkg/controller/infrastructure/infraflow/utils.go:175]: Very small opinionated remark:
I prefer the simple declaration `var res []T` instead of `make` because it does not allocate memory and res points to nil, while `res := make([]T, 0)` allocates memory and s points to memory to a slice with 0 elements.

File: pkg/controller/infrastructure/infraflow/utils_test.go
Comment (hebelsan) [inline pkg/controller/infrastructure/infraflow/utils_test.go:16]: To fully test FindExisting() shouldn't we also add a test with a selector?

**Review by hebelsan — COMMENTED** [review]
/lgtm, just two small comments

File: pkg/controller/infrastructure/infraflow/utils_test.go
Comment (kon-angelo) [reply pkg/controller/infrastructure/infraflow/utils_test.go:17]: Addressed in latest commit

**Review by AndreasBurger — APPROVED** [review]
lgtm

**Pull request merged**
